### ISSUE TITLE
[bug]: Noise threshold is not restored from saved images in the GUI

### ISSUE THREAD

**Issue by JPPhoto** [invoke-ai/InvokeAI#2049]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### OS

Linux

### GPU

cuda

### VRAM

12GB

### What happened?

When using the GUI, using all parameters from an existing InvokeAI image whose noise threshold is zero does not restore that noise threshold back to 0. This can be duplicated by opening the GUI, changing Noise Threshold to some amount, selecting an image, and pressing the Use All (A) button (or right-clicking on an image in the gallery and selecting Use All Parameters).

### Additional context

I think the following should fix it:
```
$ git diff
diff --git a/frontend/src/features/options/store/optionsSlice.ts b/frontend/src/features/options/store/optionsSlice.ts
index 65b68586..012da05d 100644
--- a/frontend/src/features/options/store/optionsSlice.ts
+++ b/frontend/src/features/options/store/optionsSlice.ts
@@ -304,8 +304,11 @@ export const optionsSlice = createSlice({
       if (sampler) state.sampler = sampler;
       if (steps) state.steps = steps;
       if (cfg_scale) state.cfgScale = cfg_scale;
-      if (threshold) state.threshold = threshold;
-      if (typeof threshold === 'undefined') state.threshold = 0;
+      if (typeof threshold === 'undefined') {
+        state.threshold = 0;
+      } else {
+        state.threshold = threshold;
+      }
       if (perlin) state.perlin = perlin;
       if (typeof perlin === 'undefined') state.perlin = 0;
       if (typeof seamless === 'boolean') state.seamless = seamless;
```

**Comment by JPPhoto** [comment]
Doesn't `if (var)` result in false if var is set and is 0? We should probably use the `typeof` logic for every parameter.

**Comment by JPPhoto** [comment]
Those changes were merged in. I wonder if the metadata is making it into the files. I'll ping some of the frontend devs.

**Comment by psychedelicious** [comment]
Hi @n4phth4 

We've fixed the bug, the fix will be in v2.3.0.
